src/sentry/api/endpoints/organization_projects.py

```diff
@@ -79,11 +83,26 @@ def get(self, request, organization):
                 teams__in=team_list,
             ).prefetch_related('teams')
 
+        query = request.GET.get('query')
+        if query:
+            tokens = tokenize_query(query)
+            for key, value in six.iteritems(tokens):
+                if key == 'query':
+                    value = ' '.join(value)
+                    queryset = queryset.filter(Q(name__icontains=value) | Q(slug__icontains=value))
+                else:
+                    queryset = queryset.none()
+        queryset = queryset.distinct()
+
         return self.paginate(
             request=request,
             queryset=queryset,
             order_by='slug',
             on_results=lambda x: serialize(x, request.user, ProjectSummarySerializer(
+                environment_id=self._get_environment_id_from_request(
+                    request,
+                    organization.id,
+                ),
```

Comment:
Heads up that this may raise `Environment.DoesNotExist` if the user passes some garbage here.

Reply:
yeah i ignored it as I figured i'd just let it error out since either way its a failure

can always follow up and patch it later